kyuubi-common/src/main/scala/org/apache/kyuubi/service/authentication/JdbcAuthenticationProviderImpl.scala

```diff
@@ -0,0 +1,235 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.service.authentication
+
+import java.sql.{Connection, PreparedStatement, Statement}
+import java.util.Properties
+import javax.security.sasl.AuthenticationException
+
+import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
+import org.apache.commons.lang3.StringUtils
+
+import org.apache.kyuubi.Logging
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf._
+
+class JdbcAuthenticationProviderImpl(conf: KyuubiConf) extends PasswdAuthenticationProvider
+  with Logging {
+
+  private val driverClass = conf.get(AUTHENTICATION_JDBC_DRIVER)
+  private val jdbcUrl = conf.get(AUTHENTICATION_JDBC_URL)
+  private val jdbcUsername = conf.get(AUTHENTICATION_JDBC_USERNAME)
+  private val jdbcUserPassword = conf.get(AUTHENTICATION_JDBC_PASSWORD)
+  private val authQuerySql = conf.get(AUTHENTICATION_JDBC_QUERY)
+
+  private val SQL_PLACEHOLDER_REGEX = """\$\{.+?}""".r
+  private val USERNAME_SQL_PLACEHOLDER = "${username}"
+  private val PASSWORD_SQL_PLACEHOLDER = "${password}"
+
+  checkJdbcConfigs()
+
+  private[kyuubi] val hikariDataSource = getHikariDataSource()
+
+  /**
+   * The authenticate method is called by the Kyuubi Server authentication layer
+   * to authenticate users for their requests.
+   * If a user is to be granted, return nothing/throw nothing.
+   * When a user is to be disallowed, throw an appropriate [[AuthenticationException]].
+   *
+   * @param user     The username received over the connection request
+   * @param password The password received over the connection request
+   * @throws AuthenticationException When a user is found to be invalid by the implementation
+   */
+  @throws[AuthenticationException]
+  override def authenticate(user: String, password: String): Unit = {
+    if (StringUtils.isBlank(user)) {
+      throw new AuthenticationException(s"Error validating, user is null" +
+        s" or contains blank space")
+    }
+
+    if (StringUtils.isBlank(password)) {
+      throw new AuthenticationException(s"Error validating, password is null" +
+        s" or contains blank space")
+    }
+
+    var connection: Connection = null
+    var queryStatement: PreparedStatement = null
+
+    try {
+      connection = hikariDataSource.getConnection
+
+      queryStatement = getAndPrepareQueryStatement(connection, user, password)
+
+      val resultSet = queryStatement.executeQuery()
+
+      if (resultSet == null || !resultSet.next()) {
+        // Auth failed
+        throw new AuthenticationException(s"Password does not match or no such user. user:" +
+          s" $user , password length: ${password.length}")
+      }
```

Comment:
is the `resultSet` leak?

Reply:
dont think so? 
ref to doc of `java.sql.Statement`, `When a Statement object is closed, its current ResultSet object, if one exists, is also closed.`

Reply:
Then it's fine, thanks for clarification

Reply:
cc @turboFei, we can add a `JdbcUtils` in common modules to share some common codes, since Jdbc is used in several places.